### PR TITLE
Fix parsing of recursive JAR URIs in `CloseablePath`

### PR DESCRIPTION
**Pull request by mpkorstanje**
## Overview

Jar uris follow the format[1]:

```
jar:<url>!/[<entry>]
```

So splitting should be done on the last `!/` rather than the first.

Fixes: #1724 for Spring Boot 3.2 and later.

 1. https://docs.oracle.com/javase/8/docs/api/java/net/JarURLConnection.html

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

**Comment by mpkorstanje** [comment]
Verified end-2-end using https://github.com/mpkorstanje/junit-5-spring-boot-3.2.0.

**Comment by sbrannen** [comment]
Thanks for the PR, @mpkorstanje. 👍 

Any chance you want to make nested JAR scanning work for Spring Boot versions prior to Spring Boot 3.2 as well (in a separate PR)? 😎

**Comment by mpkorstanje** [comment]
> Any chance you want to make nested JAR scanning work for Spring Boot versions prior to Spring Boot 3.2 as well (in a separate PR)? 😎

The general case? No chance. :smile:

Spring Boot 3.1 has [only 6 months of OSS support left](https://spring.io/projects/spring-boot#support). For the amount of effort involved, it would be gilding a _dead_ lilly.

The specific case of supporting scanning of `/BOOT-INF/classes` isn't too difficult. That mostly depends on how much of a code blister you're willing to accept.  I've put a proposal in https://github.com/junit-team/junit5/issues/1724#issuecomment-1859291992. If that doesn't look too bad, I could polish it up into a PR.

File: documentation/src/docs/asciidoc/release-notes/release-notes-5.11.0-M1.adoc [hunk 1]
```diff
@@ -19,6 +19,9 @@ repository on GitHub.
   `junit.platform.launcher.interceptors.enabled` is set to `true`.
   - See link:https://github.com/junit-team/junit5/issues/3561[issue 3561] for details.
```
Comment (marcphilipp) [inline documentation/src/docs/asciidoc/release-notes/release-notes-5.11.0-M1.adoc]:
```suggestion
```

File: junit-platform-commons/src/main/java/org/junit/platform/commons/util/CloseablePath.java [hunk 2]
```diff
@@ -53,9 +53,11 @@ static CloseablePath create(URI uri) throws URISyntaxException {
 
 	static CloseablePath create(URI uri, FileSystemProvider fileSystemProvider) throws URISyntaxException {
 		if (JAR_URI_SCHEME.equals(uri.getScheme())) {
-			String[] parts = uri.toString().split(JAR_URI_SEPARATOR);
-			String jarUri = parts[0];
-			String jarEntry = parts[1];
+			// Parsing: jar:<url>!/[<entry>]
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/util/CloseablePath.java:56]:
```suggestion
			// Parsing: jar:<url>!/[<entry>], see java.net.JarURLConnection
```

**Review by marcphilipp — APPROVED** [review]

**Comment by marcphilipp** [comment]
@mpkorstanje Thanks! 👍

**Pull request merged**